api/v1alpha1/nutanix_clusterconfig_types.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// NutanixSpec defines the desired state of NutanixCluster.
+type NutanixSpec struct {
+	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// host can be either DNS name or ip address
+	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// Nutanix Prism Central endpoint configuration.
+	PrismCentralEndpoint NutanixPrismCentralEndpointSpec `json:"prismCentralEndpoint"`
+}
+
+func (NutanixSpec) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Description: "Nutanix cluster configuration",
+			Type:        "object",
+			Properties: map[string]clusterv1.JSONSchemaProps{
+				"controlPlaneEndpoint": ControlPlaneEndpointSpec{}.VariableSchema().OpenAPIV3Schema,
+				"prismCentralEndpoint": NutanixPrismCentralEndpointSpec{}.VariableSchema().OpenAPIV3Schema,
```

Comment:
WDYT, keeping this as a "string" with validating URI format. "<host>:<port>" format for both controlplane and PC endpoints. The port can be defaulted if not provided in the URI.

Reply:
IMO I think because `NutanixPrismCentralEndpointSpec` already has other fields (Insecure, AdditionalTrustBundle, Credentials) and CAPX actually expects the DNS or IP without a scheme, we don't gain much having it a single field and then having it to do manipulation in code.

Reply:
Oh I misread that a little bit, added some validation in b75d42f52413919ccac44b2493fbb09eb4200a52

Reply:
Thanks Dimitri. you are right about PC endpoint. Thanks for adding validation and default ports